### PR TITLE
[Editor] Support dragging & dropping images on a pdf (bug 1900907)

### PR DESCRIPTION
**Pull request by calixteman**

### CODE REVIEW

**Comment by Snuffleupagus** [comment]
Something seems to be wrong with the bots, since the preview doesn't work:
```
Loading module from “http://54.193.163.58:8877/dcb16dff92c22e3/build/pdf.mjs” was blocked because of a disallowed MIME type (“application/octet-stream”).
[viewer.html](http://54.193.163.58:8877/dcb16dff92c22e3/web/viewer.html)
Loading failed for the module with source “http://54.193.163.58:8877/dcb16dff92c22e3/build/pdf.mjs”. [viewer.html:32:46](http://54.193.163.58:8877/dcb16dff92c22e3/web/viewer.html)
Loading module from “http://54.193.163.58:8877/dcb16dff92c22e3/web/viewer.mjs” was blocked because of a disallowed MIME type (“application/octet-stream”).
[viewer.html](http://54.193.163.58:8877/dcb16dff92c22e3/web/viewer.html)
Loading failed for the module with source “http://54.193.163.58:8877/dcb16dff92c22e3/web/viewer.mjs”. [viewer.html:36:42](http://54.193.163.58:8877/dcb16dff92c22e3/web/viewer.html)
```

**Comment by calixteman** [comment]
I rebooted the linux bot which was down because of the chrome crash yesterday.
About the mimetype issue on the windows bot: I don't know what's wrong.

File: web/app.js [hunk 1]
```diff
@@ -670,18 +670,28 @@ const PDFViewerApplication = {
...
+        }
       });
       appConfig.mainContainer.addEventListener("drop", function (evt) {
-        evt.preventDefault();
-
         const { files } = evt.dataTransfer;
-        if (!files || files.length === 0) {
+        if (
+          !files ||
+          files.length === 0 ||
```
Comment (Snuffleupagus) [inline web/app.js:688]: Nit: Since you're touching this anyway, I'd just combine these checks.
```suggestion
          !files?.length ||
```

**Review by Snuffleupagus — APPROVED** [review]
r=me, thank you!

File: web/app.js (hunk 1 above)
Comment (nicolo-ribaudo) [reply web/app.js:688]: You can combine even more, to `if (files?.[0].type !== "application/pdf")`

**Pull request merged**